spezi-invitation-code/index.js

```diff
@@ -0,0 +1,72 @@
+//
+// This source file is part of the StudyApplication based on the Stanford Spezi Template Application project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University
+//
+// SPDX-License-Identifier: MIT
+//
+
+const admin = require("firebase-admin");
+const {https} = require("firebase-functions/v2");
+const {FieldValue} = require("firebase-admin/firestore");
+
+class InvitationCodeVerifier {
+  constructor(firestore) {
+    this.firestore = firestore;
+  }
+
+  async verifyInvitationCode(invitationCode, userId) {
+    try {
+      const invitationCodeRef = this.firestore.doc(`invitationCodes/${invitationCode}`);
```

Comment:
I am wondering what's the advantage of passing in the firestore here compared to the creating it using the Firebase Admin SDK? It might still make sense to add some validation around userId and invitation code?

E.g., how about something like this?
```suggestion
      if (!userId) {
        throw new https.HttpsError(
            "unauthenticated",
            "User is not properly authenticated.",
        );
      }
    
      const firestore = admin.firestore();
      
      const invitationCodeRef = this.firestore.doc(`invitationCodes/${invitationCode}`);
```

Reply:
Yeah, I guess we could just spin up the Firestore object within the constructor or `enrollUserInStudy`. I thought accepting a Firestore object might make sense if someone wants to configure it in a specific way before running verification. However, it doesn't look like `admin.firestore()` really needs to take any options, so I can change the code to initialize the Firestore object rather than taking it in as a parameter in the constructor.